modules/puppet_connect/plans/test_input_data.pp

```diff
@@ -0,0 +1,31 @@
+# @summary
+#   Tests that the provided Puppet Connect input data is complete, meaning that all consuming inventory targets are connectable.
+#
+# This plan should only be used as part of the copy-pastable "test input data"
+# workflow specified in the Puppet Connect docs.
+#
+# @param targets
+#   The set of targets to test. Usually this should be 'all', the default.
+#
+# @return ResultSet the result of invoking the 'is connectable?' query on all
+# the targets. Note that this query currently consists of running the 'whoami'
+# command.
+#
+plan puppet_connect::test_input_data(TargetSpec $targets = 'all') {
+  $targs = get_targets($targets)
+  $targs.each |$target| {
+    if $target.transport != 'ssh' and $target.transport != 'winrm' {
+      fail_plan("Inventory contains target ${target} with unsupported transport, must be ssh or winrm")
+    }
+    if $target.transport == 'ssh' {
+      # Disable SSH autoloading to prevent false positive results
+      # (input data is wrong but target is still connectable due
+      # to autoloaded config)
+      set_config($target, ['ssh', 'load-config'], false)
+    }
+  }
+  # The SSH/WinRM transports will report an 'unknown host' error for targets where
+  # 'host' is unknown so run_command's implementation will take care of raising that
+  # error for us.
+  return run_command('whoami', $targs)
```

Comment:
nit, but in orch we do https://github.com/puppetlabs/orchestrator/blob/760c1f665101ee20247e021dbd24f828ec004cb2/src/puppetlabs/orchestrator/connection_tests.clj#L41-L51 

perhaps we should remain consistent? `echo 'connected'`?

Reply:
Well, `whoami` was chosen b/c we think it'll be on all SSH and Windows targets. Can the same be said for echo?

Reply:
Looks like echo exists on windows too: https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/echo

Either is likely fine, whoami might be good b/c it doesn't require params so any nasty shell escaping bugs can be avoided in windows land.

Reply:
I am fine with changing this to echo if we can agree that the probability of encountering shell escaping bugs is 0%. Otherwise, I'd like to use `whoami`.

Reply:
If we use echo, we should avoid any punctuation and only print a single word. `echo Connected` would work. Anything else gets into either shell issues (bash vs. powershell) or differences in the implementations (`echo Successfully connected` will print on two lines in powershell).

If we go with `whoami` we should just make sure the doc about the workflow explains what the expected output is.

Reply:
Thanks, updated to `echo Connected`.

Reply:
Updated the workflow doc.